dist/origin-web-common-services.js

```diff
@@ -1652,16 +1652,15 @@ angular.module("openshiftCommonServices")
       return secret;
     };
 
-    var makeBinding = function(serviceInstance, application, parametersSecretName) {
+    var makeBinding = function(serviceInstance, application, parametersSecretName, metadata) {
       var instanceName = serviceInstance.metadata.name;
 
+      metadata = _.assign({ generateName: instanceName + '-' }, metadata);
```

Comment:
Consider making a copy `angular.copy(metadata)`?

Reply:
Also, `metadata` for what?  Should probably call it `bindingMetadata`. 

However, we are passing in `serviceInstance` and `application`, full objects, seems strange not to just pass in a full `binding` object.

Reply:
> Consider making a copy angular.copy(metadata)?

I believe `_.assign` will copy the properties onto the new target object `{ generateName: instanceName + '-' }`

> Also, metadata for what? Should probably call it bindingMetadata

Since the function signature is `makeBinding` i think it's implied that `metadata` is going to be part of the binding but I can update for clarity if you think it needs it?

> However, we are passing in serviceInstance and application, full objects, seems strange not to just pass in a full binding object.

We are creating the binding here so the only extra config needed is `parametersSecretName, metadata`

Reply:
Right, sorry, just thinking out loud a bit.

That is true on the `makeBinding()` name, so prob no point in being excessively verbose.  I may have no issue here, actually.  

Thanks!